symengine/dict.cpp

```diff
@@ -204,6 +210,21 @@ bool umap_basic_basic_eq(const umap_basic_basic &a,
     return true;
 }
 
+#if defined(HAVE_SYMENGINE_PIRANHA)
+bool hash_set_eq(const hash_set &a,
+        const hash_set &b)
+{
+    // Can't be equal if # of elements in set differ:
+    if (a.size() != b.size()) return false;
+    // Loop over elements in "a":
+    for (auto &p: a) {
+        auto f = b.find(p);
+        if (f == b.end()) return false; // no such element in "b"
+    }
+    return true;
+}
+#endif
```

Comment:
Since #533 was merged now, I would implement
```c++
template<class T>
bool uset_eq(const T &a, const T &b) 
```
That works for all kinds of sets (including the `std::unordered_set`).